extension-spring/src/test/java/test/io/smallrye/openapi/runtime/scanner/entities/GreetingParam.java

```diff
@@ -0,0 +1,16 @@
+package test.io.smallrye.openapi.runtime.scanner.entities;
+
+import jakarta.ws.rs.QueryParam;
+
+public class GreetingParam {
+    @QueryParam("nameQuery") // "real" spring does not require this, but quarkus-spring-web does
```

Comment:
Does this mean neither the OpenAPI nor the actual web application work correctly without using the Jakarta REST `@QueryParam` annotation in this way?

Reply:
yes that is right

@phillip-kruger this is the reason for having `JAKARTA_QUERY_PARAM` in `SpringParameter.java`

Reply:
i did play around with https://github.com/flyinfish/spring-web-quickstart/compare/feature/adapt-to-patch?expand=1

`HelloParamCheatedWithQueryParam` has the same role there as `GreetingParm` above

Reply:
That does not seem right. At least the actual endpoint should work, even if openapi can not generate correctly. @geoand ?

Reply:
AFAIR, we don't support Spring's ParamObject, although we certainly could

Reply:
The issue (I haven't tried using Spring in Quarkus myself) is that using `GreetingParam` as a REST method parameter on a `GET` request doesn't automatically take the bean's fields to be query parameters without adding a Jakarta annotation to the fields. If that's how Spring really behaves (I don't know), then I think both Quarkus and the OpenAPI scanner should understand that whether the springdoc `@ParameterObject` annotation is present or not.

Reply:
Agree!

Reply:
i agree that `@ParameterObject` is only for documentation.

there is an issue that spring obviously does not depend on jakarta-query-param beeing present on query-class. but quarkus does - so what is your suggestion?

Reply:
Hi @MikeEdgar @phillip-kruger 

can you please approve or say what more you need?

in a perfect situation quarkus would support objects on get-methods as spring does. since quarkus needs @jakarka-query-param, this pr is imho a suitable solution as it supports on openapi-docs the features quarkus support at runtime at least as far as i understand it.

were migrating like one hundred spring-microservices to quarkus and would be very happy to do it with the least possible change of code, which means migrating to quarkus-spring-web rather than "pure" quarkus.

thank you very much

Reply:
@geoand - can you comment on this comment: https://github.com/smallrye/smallrye-open-api/pull/1823#discussion_r1592151911

Reply:
I don't remember how Spring behaves in this case, I would have to check and unfortunately I am short on time at the moment

Reply:
@MikeEdgar if we were talking spring i would agree shat scanner should not use jakarta-annotations.

since we are talking quarkus (do we? is this project used outside quarkus?) i would not be too happy if the scanner generates an api which then does not work until i add the missing jakarta-annotations.

Reply:
This is being used in many other projects, not only Quarkus. I agree with @MikeEdgar, the fix should be that we handle the un-annotated case.

Reply:
I've re-based this one and pushed another commit to remove the reliance on the Jakarta annotation for Spring. As we discussed earlier in this thread, Quarkus itself may still not handle this scenario (I haven't looked at the changes in Quarkus Spring since the last comment), but this at least will scan the way it should work with native Spring.

@flyinfish, would you mind reviewing the current state of this PR and let us know if it should work as needed for your OpenAPI documentation?

Reply:
@MikeEdgar 
sorry for late reply - Yes ist works as needed - Thank you very much!

nevertheless we moved away from spring-approach and convert our legacy apps now to proper quarkus/microprofile annotations.

anyhow for spring-users this should help!